### DISCUSSION TITLE
Bulk Import

### DISCUSSION THREAD

**Discussion by Vigilin** [pglombardo/PasswordPusher#260] [Q&A] [answered]
Hi Peter, first off, I think you're creating an awesome site here that will be going places!

I have a question.  I need to bulk import a hundred or so passwords and export the links.  I see you don't have something like this, but I am able to kinda import them via curl.  The only issue is that I would need them to be associated with my account.

I see that the curl output does have a "user_id" field.  Am I able to use that somehow to associate my creations on curl with my web based account?

Thanks in advance, Chris

**Comment by pglombardo** [comment]
Hi Chris - thanks for the kind words.  

Using curl or one of the command line utilities is the right approach but I haven't added CLI authentication yet.  :-(  I've been planning to add this soon but I'm not sure when yet.

There are two other potential options in the meantime:

1. Write a Javascript loop to run from within the browser console - where you are already logged in.
2. Pass the [pwpush cookies to Curl](https://stackoverflow.com/questions/12399087/curl-to-access-a-page-that-requires-a-login-from-a-different-page).

Let me know what you decide.  I'll update here once CLI authentication is added.

**Comment by pglombardo** [comment, marked as answer]
HI @Vigilin - Authentication for the JSON API has been released.  There is a short twitter thread with descriptions here: https://twitter.com/pwpush/status/1563889107536941057

Documentation for the API: https://pwpush.com/api

Also in progress but not yet released is [pwpush-cli](https://github.com/pglombardo/pwpush-cli)

![Screen Shot 2022-08-28 at 15 56 35](https://user-images.githubusercontent.com/395132/187092262-3e5c4757-89d2-495d-85a2-8930d24277b3.png)
